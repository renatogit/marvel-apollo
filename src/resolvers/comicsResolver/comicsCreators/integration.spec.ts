const DataSourcesComicsCreatorsAPI = require('@/dataSources/comics');
const ComicsCreatorsResolver = require('.');
const ServerGetComicsCreators = require('@/configs/test');
const QUERY_GET_COMICS_CREATORS = require('@query/comics/queryComicsCreators');
const MOCK_GET_COMICS_CREATORS = require('@mock/comics/mockComicsCreators');

describe('resolvers/comicsCreatorsResolver/integration', () => {
    const comicsId = { comicsId: 'fake-code' };

    it('fetches comicsCreators and generate snapshots', async () => {
        const { server } = await ServerGetComicsCreators(
            DataSourcesComicsCreatorsAPI,
            ComicsCreatorsResolver,
            MOCK_GET_COMICS_CREATORS,
            comicsId
        );

        const { body } = await server.executeOperation({
            query: QUERY_GET_COMICS_CREATORS,
            variables: comicsId,
        });

        expect(body).toMatchSnapshot();
    });
});
