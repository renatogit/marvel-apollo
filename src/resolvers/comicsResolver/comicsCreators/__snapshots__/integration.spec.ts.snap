// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/comicsCreatorsResolver/integration fetches comicsCreators and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "comicsCreators": {
        "attributionHTML": "mock data",
        "attributionText": "mock data",
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 9,
          "limit": 20,
          "offset": 0,
          "results": [
            {
              "comics": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "events": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "firstName": "mock data",
              "fullName": "mock data",
              "id": 87,
              "lastName": "mock data",
              "middleName": "mock data",
              "modified": "mock data",
              "resourceURI": "mock data",
              "series": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "stories": null,
              "suffix": "mock data",
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
            {
              "comics": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "events": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "firstName": "mock data",
              "fullName": "mock data",
              "id": 87,
              "lastName": "mock data",
              "middleName": "mock data",
              "modified": "mock data",
              "resourceURI": "mock data",
              "series": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "stories": null,
              "suffix": "mock data",
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
          ],
          "total": 9,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
